### PR TITLE
Cats and stuff

### PR DESCRIPTION
**Pull request by adelbertc**
It compiles 🤷‍♂️ It has... very minimal tests. It wasn't even using the Scalaz ScalaCheck bindings.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -2,7 +2,7 @@ name := "parser"
 
 organization in Global := "io.verizon.ermine"
 
-crossScalaVersions in Global := Seq("2.12.1", "2.11.8", "2.10.6")
+crossScalaVersions in Global := Seq("2.12.4", "2.11.12", "2.10.6")
```
Comment (rossabaker) [inline build.sbt]: We should bump these in Travis, too.  That's what will be in effect when we publish.

File: build.sbt [hunk 2]
```diff
@@ -20,13 +20,15 @@ scalacOptions ++=
 
 resolvers += Resolver.bintrayRepo("non", "maven") // for kind-projector
 
-addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.3")
+addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.4")
```
Comment (rossabaker) [inline build.sbt]: There's an 0.9.5 now.  I don't know why it's better than 0.9.4, but I just bumped it on another project.

File: src/main/scala/scalaparsers/Token.scala [hunk 3]
```diff
@@ -5,21 +5,20 @@ package scalaparsers
   * @author EAK
   */
 
-sealed abstract class Token
+sealed abstract class Token extends Product with Serializable
 
-case object VSemi extends Token {
+final case object VSemi extends Token {
```
Comment (rossabaker) [inline src/main/scala/scalaparsers/Token.scala:10]: I like the final case classes.  Is "final case object" different from "case object" in some subtle way I've not yet discovered?

**Review by rossabaker — CHANGES_REQUESTED** [review]
Mostly LGTM. I'd bump those Scala versions in Travis while we're here.

File: src/main/scala/scalaparsers/Token.scala (hunk 3 above)
Comment (adelbertc) [reply src/main/scala/scalaparsers/Token.scala:10]: No clue, I just have a habit/compulsion to tack `final` onto any `case {class,object}`. People seem to like it https://twitter.com/viktorklang/status/735556776989491201 😉

Comment (rossabaker) [reply src/main/scala/scalaparsers/Token.scala:10]: Yeah, I think it's great on case class.  I don't think it matters one way or another on case object.  Mostly asking out of curiosity.

**Comment by rossabaker** [comment]
LGTM.  Leaving the stop sign up while we wait for cats-1.0 and to get rid of the orphan instances with whatever solution it offers.

**Comment by timperrett** [comment]
`version.sbt` needs to be updated to the next major version as this release will be entirely incompatible.

**Review by rossabaker — APPROVED** [review]

**Pull request merged**